components/Carousel/Carousel.js

```diff
@@ -1,14 +1,34 @@
 class Carousel {
+    //  grab a reference to the carousel, and in it grab the left and right buttons
+    //  grab a reference to all of the images
+    constructor(carousel) {
+        this.carousel = carousel;
+        this.images = document.querySelectorAll('.carousel img')
+        this.imagesmap = Array.from(this.images)
+        this.current = this.images[0]; // Create a current index
+        this.current.style.display = 'inline-block';
+        this.carousel.addEventListener('click', () => {this.arrowSelection()}  // buttons are gonna need some click handlers.
+        );
+    }
+    arrowSelection() {
```

Comment:
Loved the implementation. if you have looked up this thing from internet. Please comment the reference.